pandera/schemas.py

```diff
@@ -598,6 +610,23 @@ def _validate(
                                 check="column_ordered",
                             ),
                         )
+        if self._allow_duplicate_column_names is False:
+            failed = check_obj.columns[check_obj.columns.duplicated()]
+            if failed.any():
+                msg = (
+                    "dataframe contains multiple columns with label(s): "
+                    f"{failed.tolist()}"
+                )
+                error_handler.collect_error(
+                    "duplicate_dataframe_column_labels",
+                    errors.SchemaError(
+                        self,
+                        check_obj,
+                        msg,
+                        failure_cases=reshape_failure_cases(pd.Series(failed)),
```

Comment:
reporting non-scalar failure cases only really applies to checks that involve validating actual column/index values. Since this check validates metadata about the dataframe (i.e. column names) it should be reported as a scalar (see line 597).

```python
failure_cases=scalar_failure_case(failed)
```

Reply:
Thanks, I knew this wasn't quite right but wasn't sure exactly how to handle it.